src/lib.rs

```diff
@@ -250,6 +270,32 @@ where
         Ok(())
     }
 
+    pub fn cad(
+        &mut self,
+        delay: &mut dyn DelayMs<u8>
+    ) -> Result<usize, Error<E, CS::Error, RESET::Error>>
+    {
+        self.set_mode(RadioMode::Stdby)?;
+        self.set_dio(FlagsDIO::DIO0_CAD_DONE, FlagsDIO::DIO0_GROUP)?;
+        self.set_dio(FlagsDIO::DIO1_CAD_DETECTED, FlagsDIO::DIO1_GROUP)?;
+        /* enable all interrupts */
+        self.write_register(Register::RegIrqFlagsMask.addr(), 0x00)?;
+        self.write_register(Register::RegIrqFlags.addr(), 0x05)?;
+        self.set_mode(RadioMode::CAD)?;
+        Ok(1)
+    }
+
+    pub fn sleep(
```

Comment:
This looks good. 2 questions:

- would it be useful to be able to tell if the module is currently in sleep mode (something like `radio.is_sleep()`?
- is there a way how to bring it back from sleep (possibly with the same register settings as there where before sleep)?

Reply:
> 

- would it be useful to be able to tell if the module is currently in sleep mode (something like radio.is_sleep()?

I don't see any reason for that. See my comment below ...

> 

- is there a way how to bring it back from sleep (possibly with the same register settings as there where before sleep)?

If I understand to documentation correctly, that changes of LoRa mode are through "mode changing" from Stand-By mode. And after then follows some specific type of configuration for this new mode (for Rx, Tx, or CAD). This means that a new mode configuration is always necessary.